pkg/serving/config_changes.go

```diff
@@ -35,34 +37,59 @@ func UpdateEnvVars(template *servingv1alpha1.RevisionTemplateSpec, vars map[stri
 	return nil
 }
 
-// Update min and max scale annotation if larger than 0
-func UpdateConcurrencyConfiguration(template *servingv1alpha1.RevisionTemplateSpec, minScale int, maxScale int, target int, limit int) {
-	if minScale != 0 {
-		UpdateAnnotation(template, "autoscaling.knative.dev/minScale", strconv.Itoa(minScale))
-	}
-	if maxScale != 0 {
-		UpdateAnnotation(template, "autoscaling.knative.dev/maxScale", strconv.Itoa(maxScale))
-	}
-	if target != 0 {
-		UpdateAnnotation(template, "autoscaling.knative.dev/target", strconv.Itoa(target))
+// UpdateMinScale updates min scale annotation
+func UpdateMinScale(template *servingv1alpha1.RevisionTemplateSpec, min int) error {
+	return UpdateAnnotation(template, autoscaling.MinScaleAnnotationKey, strconv.Itoa(min))
+}
+
+// UpdatMaxScale updates max scale annotation
+func UpdateMaxScale(template *servingv1alpha1.RevisionTemplateSpec, max int) error {
+	return UpdateAnnotation(template, autoscaling.MaxScaleAnnotationKey, strconv.Itoa(max))
+}
+
+// UpdateConcurrencyTarget updates container concurrency annotation
+func UpdateConcurrencyTarget(template *servingv1alpha1.RevisionTemplateSpec, target int) error {
+	// TODO(toVersus): Remove the following validation once serving library is updated to v0.8.0
+	// and just rely on ValidateAnnotations method.
+	if target < autoscaling.TargetMin {
+		return fmt.Errorf("Invalid %s annotation value: must be an intger greater than 0", autoscaling.TargetAnnotationKey)
```

Comment:
*integer

Reply:
How about printing error message as
```
("Invalid `concurrency-target` value: %d , must be an integer > 0", target)
```
easy for user to know which flag value to be adjusted than printing the annotation key
WDYT ?

Reply:
Yeah, that’s more like it! Thanks.